test/evp_test.c

```diff
@@ -1258,6 +1259,13 @@ static int pkey_test_ctrl(EVP_PKEY_CTX *pctx, const char *value)
     if (p != NULL)
         *p++ = 0;
     rv = EVP_PKEY_CTX_ctrl_str(pctx, tmpval, p);
+    if (p != NULL && rv <= 0 && rv != -2) {
```

Comment:
There's a check to see if the value has an OID, and only skip the test then.  It would be quite an advanced typo to misspell it into another (disabled) algo.

The point is that if an algo is disabled, for example MDC2, this allows a test with `Ctrl = digest:MDC2` to be skipped rather than failing.  We have a test like that in `evptests.txt`, and test_evp currently fails when configured `no-mdc2` (or `no-des`) because of that...

We have a similar check for disabled algos in `digest_test_init()`, that's where I got my inspiration.